### ISSUE TITLE
Programmatically swipe an item ?

### ISSUE THREAD

**Issue by GauthierRoomDotMe** [h6ah4i/android-advancedrecyclerview#155]
Is there a way to programmatically swipe from unpin to pin state ?

What I want to achieve is to let the user click an unpinned item, having an animation that swipe the item to left for half its size and then if the user click again on the item it swipes to the unpin state.

Thanks to the example I achieved to the later part by simply using item.setPinned(false); and notifying the adapter. It animates nicely.

But when I do item.setPinned(true); or holder.setSwipeItemHorizontalSlideAmount(-0.5f); it directly goes to the pin state without animation.

Is there a way to do this animation programmatically ?

**Comment by h6ah4i** [comment]
Hi. Unfortunately, it requires adding a new method to update item pinned state with animation :( I will add it, but it takes time because I have been tied up with work...

**Comment by mdvoretskiy** [comment]
Same problem appears when item automatically goes to unpined state when swiping ("Button under swipeable item" in demo).

**Comment by jackellenberger** [comment]
@GauthierRoomDotMe To clarify, does holder.setSwipeItemHorizontalSlideAmount(-0.5f); actually do anything for you? It seems like my mHorizontalSlideAmount gets updated, but nothing actually moves.

**Comment by GauthierRoomDotMe** [comment]
@jackellenberger Yes holder.setSwipeItemHorizontalSlideAmount(-0.5f); works fine with me.

Do you set holder.setMaxLeftSwipeAmount(MAX_LEFT_SWIPE); before calling it ? it must be equal or less than -0.5f in our case.

**Comment by jackellenberger** [comment]
I never ended up getting that working (looking at it now, I think it was because I was using GroupSwipeRightResultAction instead of Left or something), but instead set up an animation:

`
holder.mContainer.animate().translationX(-holder.mContainer.getWidth())
                            .setDuration(mShortAnimationDuration)
                            .setListener(removerListener);
`

removerListener is an AnimatorListener that deletes the item after it has been swiped.
